pkg/ast/ast_union_type_extension.go

```diff
@@ -37,18 +37,35 @@ func (d *Document) UnionTypeExtensionHasDirectives(ref int) bool {
 	return d.UnionTypeExtensions[ref].HasDirectives
 }
 
-func (d *Document) ExtendUnionTypeDefinitionByUnionTypeExtension(unionTypeDefinitionRef, unionTypeExtensionRef int) {
+func (d *Document) ExtendUnionTypeDefinitionByUnionTypeExtension(unionTypeDefinitionRef, unionTypeExtensionRef int) (string, string) {
 	if d.UnionTypeExtensionHasDirectives(unionTypeExtensionRef) {
 		d.UnionTypeDefinitions[unionTypeDefinitionRef].Directives.Refs = append(d.UnionTypeDefinitions[unionTypeDefinitionRef].Directives.Refs, d.UnionTypeExtensions[unionTypeExtensionRef].Directives.Refs...)
 		d.UnionTypeDefinitions[unionTypeDefinitionRef].HasDirectives = true
 	}
 
 	if d.UnionTypeExtensionHasUnionMemberTypes(unionTypeExtensionRef) {
-		d.UnionTypeDefinitions[unionTypeDefinitionRef].UnionMemberTypes.Refs = append(d.UnionTypeDefinitions[unionTypeDefinitionRef].UnionMemberTypes.Refs, d.UnionTypeExtensions[unionTypeExtensionRef].UnionMemberTypes.Refs...)
-		d.UnionTypeDefinitions[unionTypeDefinitionRef].HasUnionMemberTypes = true
+		memberSet := make(map[string]bool)
```

Comment:
I talked to Sergey about this code, and we probably want to revert this code and keep it remain naïve. The validation of the union members would then occur in the validation package (but as far as I can tell, this validation is currently missing).

However, when Wojciech used `go generate ./..` with a union that had a duplicate member, it appeared to fail.